cirq-web/cirq_ts/src/bloch_sphere/components/vector.ts

```diff
@@ -0,0 +1,95 @@
+// Copyright 2021 The Cirq Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {ArrowHelper, Vector3, Group, LineBasicMaterial} from 'three';
+
+/**
+ * Generates a vector to add to the Bloch sphere.
+ * The length and direction of the vector are configurable.
+ */
+export class Vector extends Group {
+  readonly scaling_factor: number;
+  readonly x: number;
+  readonly y: number;
+  readonly z: number;
+
+  /**
+   * Class constructor.
+   * @param x the x coordinate of the vector tip
```

Comment:
It is a bit still unclear to me from this dosctring what the x,y,z are for. Describe what exactly these are...or maybe we can rename the class to `ScaledBlochVector` or `StateVector` and then it become a bit clearer, e.g. "x,y,z provide the coordinates of the state vector within the Bloch sphere with a unit radius. However, as the actual sphere's radius can be any ThreeJS unit, we can scale the vector along with it, by specifying `bloch_sphere_radius`" -- and then I'd rename the scaling_factor to `bloch_sphere_radius`.

Reply:
Changed to StateVector and added clarification